inkcpp/include/globals.h

```diff
@@ -4,15 +4,43 @@
 
 namespace ink::runtime
 {
+	template<typename T>
+	struct ret {
+		operator T() { return value; }
+		operator bool() { return has_value; }
+		T value;
+		bool has_value;
+	};
 	/**
 	* Represents a global store to be shared amongst ink runners.
 	* Stores global variable values, visit counts, turn counts, etc.
 	*/
-	class globals_interface
+	class globals
 	{
 	public:
 		// No public interface yet
 		virtual void dummy() = 0;
-		virtual ~globals_interface() = default;
+		virtual ~globals() = default;
+
+		template<typename T>
+		ret<T> get(const char* name) {
+			static_assert(internal::always_false<T>::value,
+					"No getter for his type!");
+		}
+
+	protected:
+		virtual int getInt(const char* name) = 0;
```

Comment:
Agree these might seem ugly but I think they're the price of hiding everything behind interfaces. I have no issue with them; it's how I would have done it.